src/Microsoft.OData.Core/SelectedPropertiesNode.cs

```diff
@@ -204,6 +210,19 @@ internal enum SelectionType
             PartialSubtree = 2,
         }
 
+        /// <summary>
+        /// Gets names of the expanded entity nodes at current level.
+        /// </summary>
+        internal IEnumerable<string> NamesOfSelectedExpandedEntities

```

Comment:
>Selected [](start = 44, length = 8)

This name implies that the function returns the names of selected as well as expanded entities.  I think it just returns the expanded entities.  Also, I don't think it adds much to put "NamesOf" in the name, and doing so is not a pattern we follow other places, so I'd just call this ExpandedEntities.

Reply:
renamed to SelectedExpandedEntities. What we need is the expanded entities' names from the keys, instead from the value objects.

---
In reply to: [229464026](https://github.com/OData/odata.net/pull/1106#discussion_r229464026) [](ancestors = 229464026)